File: app/src/main/java/com/krystianwsul/checkme/gui/DebugFragment.java
package com.krystianwsul.checkme.gui;


import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.krystianwsul.checkme.DataDiff;
import com.krystianwsul.checkme.MyCrashlytics;
import com.krystianwsul.checkme.R;
import com.krystianwsul.checkme.domainmodel.DomainFactory;
import com.krystianwsul.checkme.notifications.TickService;
import com.krystianwsul.checkme.utils.time.ExactTimeStamp;

import junit.framework.Assert;

public class DebugFragment extends AbstractFragment {
    @NonNull
    public static DebugFragment newInstance() {
        return new DebugFragment();
    }

    public DebugFragment() {

    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_debug, container, false);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        View view = getView();
        Assert.assertTrue(view != null);

        Button debugException = view.findViewById(R.id.debug_exception);
        Assert.assertTrue(debugException != null);

        debugException.setOnClickListener(v -> {
            int i = 0;
            @SuppressWarnings("unused") int j = 1 / i;
        });

        Button debugTick = view.findViewById(R.id.debug_tick);
        Assert.assertTrue(debugTick != null);

        debugTick.setOnClickListener(v -> TickService.Companion.startServiceDebug(getActivity(), "DebugFragment: TickService.startServiceDebug"));

        TextView debugData = view.findViewById(R.id.debug_data);
        Assert.assertTrue(debugData != null);

        Button debugLoad = view.findViewById(R.id.debug_load);
        Assert.assertTrue(debugLoad != null);

        debugLoad.setOnClickListener(v -> {
            StringBuilder stringBuilder = new StringBuilder();

            SharedPreferences sharedPreferences = getActivity().getSharedPreferences(TickService.Companion.getTICK_PREFERENCES(), Context.MODE_PRIVATE);
            long lastTick = sharedPreferences.getLong(TickService.Companion.getLAST_TICK_KEY(), -1);
            String tickLog = sharedPreferences.getString(TickService.Companion.getTICK_LOG(), "");

            ExactTimeStamp lastTickExactTimeStamp = new ExactTimeStamp(lastTick);

            stringBuilder.append("last beeping tick: ");
            stringBuilder.append(lastTickExactTimeStamp.toString());
            stringBuilder.append("\ntick log:\n");
            stringBuilder.append(tickLog);

            DomainFactory domainFactory = DomainFactory.getDomainFactory();

            stringBuilder.append("\ndata load time: ");
            stringBuilder.append((domainFactory.getReadMillis() + domainFactory.getInstantiateMillis()));
            stringBuilder.append("ms (");
            stringBuilder.append(domainFactory.getReadMillis());
            stringBuilder.append(" + ");
            stringBuilder.append(domainFactory.getInstantiateMillis());
            stringBuilder.append(")");

            stringBuilder.append("\ntasks: ");
            stringBuilder.append(domainFactory.getTaskCount());
            stringBuilder.append(", instances: ");
            stringBuilder.append(domainFactory.getInstanceCount());
            stringBuilder.append(", custom times: ");
            stringBuilder.append(domainFactory.getCustomTimeCount());

            ExactTimeStamp t1 = ExactTimeStamp.getNow();
            DomainFactory.getDomainFactory().getGroupListData(getActivity(), ExactTimeStamp.getNow(), 0, MainActivity.TimeRange.DAY);
            ExactTimeStamp t2 = ExactTimeStamp.getNow();

            stringBuilder.append("\ntoday: ");
            stringBuilder.append((t2.getLong() - t1.getLong()));
            stringBuilder.append(" ms");

            stringBuilder.append("\ncrashlytics enabled: ");
            stringBuilder.append(MyCrashlytics.getEnabled());

            debugData.setText(stringBuilder);
        });

        Button debugDiffButton = view.findViewById(R.id.debug_diff_button);
        Assert.assertTrue(debugDiffButton != null);

        TextView debugDiffText = view.findViewById(R.id.debug_diff_text);
        Assert.assertTrue(debugDiffText != null);

        debugDiffButton.setOnClickListener(v -> debugDiffText.setText(DataDiff.getDiff()));
    }
}

File: app/src/main/java/com/krystianwsul/checkme/gui/DebugFragment.kt
package com.krystianwsul.checkme.gui


import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.krystianwsul.checkme.MyCrashlytics
import com.krystianwsul.checkme.Preferences
import com.krystianwsul.checkme.R
import com.krystianwsul.checkme.domainmodel.DomainFactory
import com.krystianwsul.checkme.notifications.TickJobIntentService
import com.krystianwsul.common.time.ExactTimeStamp
import kotlinx.android.synthetic.main.fragment_debug.*

class DebugFragment : AbstractFragment() {

    companion object {

        fun newInstance() = DebugFragment()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?) = inflater.inflate(R.layout.fragment_debug, container, false)!!

    @SuppressLint("SetTextI18n")
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        debugException.setOnClickListener {
            val i = 0
            @Suppress("DIVISION_BY_ZERO", "UNUSED_VARIABLE")
            val j = 1 / i
        }

        debugTick.setOnClickListener { TickJobIntentService.startServiceNormal(requireContext(), "DebugFragment") }

        debugLoad.setOnClickListener {
            debugData.text = StringBuilder().apply {
                val lastTick = Preferences.lastTick
                val tickLog = Preferences.tickLog.log

                val lastTickExactTimeStamp = ExactTimeStamp(lastTick)

                val domainFactory = DomainFactory.instance

                append("local load time: ")
                append(domainFactory.localReadTimes.readMillis + domainFactory.localReadTimes.instantiateMillis)
                append("ms (")
                append(domainFactory.localReadTimes.readMillis)
                append(" + ")
                append(domainFactory.localReadTimes.instantiateMillis)
                append(")")

                append("\nremote load time: ")
                append(domainFactory.remoteReadTimes.readMillis + domainFactory.remoteReadTimes.instantiateMillis)
                append("ms (")
                append(domainFactory.remoteReadTimes.readMillis)
                append(" + ")
                append(domainFactory.remoteReadTimes.instantiateMillis)
                append(")")

                domainFactory.remoteUpdateTime?.let {
                    append("\nremote update time: ")
                    append(it)
                    append("ms")
                }

                append("\n\ntasks: ")
                append(domainFactory.taskCount)
                append("\ninstances: ")
                append(domainFactory.instanceCount)
                append("\ncustom times: ")
                append(domainFactory.customTimeCount)
                append("\ninstance shown: ")
                append(domainFactory.instanceShownCount)

                append("\n\n")
                append(domainFactory.irrelevantSummary)

                val t1 = ExactTimeStamp.now
                DomainFactory.instance.getGroupListData(ExactTimeStamp.now, 0, MainActivity.TimeRange.DAY)
                val t2 = ExactTimeStamp.now

                append("\n\ntoday: ")
                append(t2.long - t1.long)
                append(" ms")

                append("\ncrashlytics enabled: ")
                append(MyCrashlytics.enabled)

                append("\n\ntemporary notification log:")
                append(Preferences.temporaryNotificationLog.log)

                append("\n\nlast beeping tick: ")
                append(lastTickExactTimeStamp.toString())
                append("\n\ntick log:\n")
                append(tickLog)
            }
        }
    }
}
